### ISSUE TITLE
Not setting permissions when creating new app

### ISSUE THREAD

**Issue by ryancbutler** [ryancbutler/XDReplicate#10]
> User Creation issue: Not solved:
> 
> ::: If it finds a folder (in this case this is the root Applications folder), it works great and it creates both the app and the user accounts associated.
> 
> However, if the folder is new, and the app must be created, it cannot figure out how to add user accounts to the app. Below is a capture of the progress. Note: If you re-run the app, it works mostly. It will fix the permissions it finds missing. I will open a new comment window for the second pass errors.
> ::::
> 
> Proccessing App Citrix AppCent
> Creating App
> Processing
> FOLDER FOUND
> Setting App Permissions
> MyDomain\My_Account
> Proccessing App ScriptedImages Push
> Creating App
> Processing TEAM Troubleshooting\Apps
> FOLDER NOT FOUND
> Creating folder
> Processing TEAM Troubleshooting
> FOLDER NOT FOUND
> Processing TEAM Troubleshooting\Apps
> FOLDER NOT FOUND
> Setting App Permissions
> MyDomain\AnotherAccount
> Add-BrokerUser : Application not found
> At \MyDomain\TEAM\scripts\citrix\XA7_replication\XDReplicate_6-27-17.ps1:485 char:17
> 
>             Add-BrokerUser -AdminAddress $xdhost -Name $user -Application $a ...
>             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> CategoryInfo : ObjectNotFound: (:) [Add-BrokerUser], SdkOperationException
> FullyQualifiedErrorId : Citrix.XDPowerShell.Broker.UnknownApplication,Citrix.Broker.Admin.SDK.AddBrokerUserCommand
> MyDomain\My_Account
> Add-BrokerUser : Application not found
> At \MyDomain\TEAM\scripts\citrix\XA7_replication\XDReplicate_6-27-17.ps1:485 char:17
> 
>             Add-BrokerUser -AdminAddress $xdhost -Name $user -Application $a ...
>             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> CategoryInfo : ObjectNotFound: (:) [Add-BrokerUser], SdkOperationException
> FullyQualifiedErrorId : Citrix.XDPowerShell.Broker.UnknownApplication,Citrix.Broker.Admin.SDK.AddBrokerUserCommand

**Comment by ryancbutler** [comment]
This should hopefully be fixed with 1.2.4.  Please test.

**Comment by Spontrel** [comment]
Outstanding Ryan.  So far, I haven't found any problems.  Nice work.  I did have to back rev all my sites to 7.12.  I found a bug in the Smart Tools Migration process and their documentation (requiring Domain Admin which was stupid) for 7.14.1 and opened a case with Citrix.  Not only did they confirm the bug is real, they decided to not only fix the documentation, but to remove all documentation suggesting to use mart Tools for 7.14.1 and limit support only to 7.12.  They agreed it was so broken, they had to remove it.  I am very glad your script is working well now.
